x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/collector/cluster/ClusterStatsMonitoringDoc.java

```diff
@@ -118,6 +125,25 @@ boolean getClusterNeedsTLSEnabled() {
         return clusterNeedsTLSEnabled;
     }
 
+    Settings getClusterMetadata() {
```

Comment:
I'm not sure this method is doing what makes the most sense.  There's a couple things:

1) Transient settings [typically override persistent settings](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-update-settings.html#_precedence_of_settings) so the order of preference should be Transient, then Persistent, then Default to match the rest of the system.
2) Despite being displayed as an object, settings are each kept as individual keys.  So you can have a situation like this:

```json
GET /_cluster/settings
{
  "persistent": {
    "cluster": {
      "metadata": {
        "thing_c": "baz",
        "thing_a": "foo"
      }
    }
  },
  "transient": {
    "cluster": {
      "metadata": {
        "thing_a": "overridden_foo",
        "thing_b": "bar"
      }
    }
  }
}
```

If these settings actually. controlled anything on the system, the cluster use these settings values:
```
cluster.metadata.thing_a = overridden_foo
cluster.metadata.thing_b = bar
cluster.metadata.thing_a = baz
```

I'm not sure whether it would be best to keep both persistent and transient settings or to merge them following the pattern used in the rest of Elasticsearch, but I think we should do one of those for consistency.